website/docs/experimental/rerun.md

```diff
@@ -0,0 +1,91 @@
+---
+id: rerun
+title: Re-run a job from previous config
+sidebar_label: Re-run 
+---
+
+import {ExampleGithubLink} from "@site/src/components/GithubLink"
+
+<ExampleGithubLink text="Example application" to="examples/experimental/rerun"/>
+
+:::caution
+This is an experimental feature. Please read through this page to understand what is supported.
+:::
+
+We use the example app linked above for demonstration. To save the configs for re-run, first use the experimental
+Hydra Callback for saving the job info:
+
+
+```yaml title="config.yaml"
+hydra:
+  callbacks:
+    save_job_info:
+      _target_: hydra.experimental.pickle_job_info_callback.PickleJobInfoCallback
+```
+
+
+
+
+```python title="Example function"
+@hydra.main(config_path=".", config_name="config")
+def my_app(cfg: DictConfig) -> None:
+    log.info(f"output_dir={HydraConfig.get().runtime.output_dir}")
+    log.info(f"cfg.foo={cfg.foo}")
+```
+
+
+Run the example app:
+```commandline
+$ python my_app.py
+[2022-03-16 14:51:30,905][hydra.experimental.pickle_job_info_callback][INFO] - Saving job configs in /Users/jieru/workspace/hydra/examples/experimental/outputs/2022-03-16/14-51-30/.hydra/config.pickle
+[2022-03-16 14:51:30,906][__main__][INFO] - Output_dir=/Users/jieru/workspace/hydra/examples/experimental/outputs/2022-03-16/14-51-30
+[2022-03-16 14:51:30,906][__main__][INFO] - cfg.foo=bar
+[2022-03-16 14:51:30,906][hydra.experimental.pickle_job_info_callback][INFO] - Saving job_return in /Users/jieru/workspace/hydra/examples/experimental/outputs/2022-03-16/14-51-30/.hydra/job_return.pickle
+```
+The Callback saves `config.pickle` in `.hydra` sub dir, this is what we will use for rerun.
+
+Now rerun the app
+```commandline
+$ OUTPUT_DIR=/Users/jieru/workspace/hydra/examples/experimental/outputs/2022-03-16/14-51-30/.hydra/
+$ python my_app.py --experimental-rerun $OUTPUT_DIR/config.pickle
+/Users/jieru/workspace/hydra/hydra/main.py:23: UserWarning: Experimental rerun CLI option.
+  warnings.warn(msg, UserWarning)
+[2022-03-16 14:59:21,666][__main__][INFO] - Output_dir=/Users/jieru/workspace/hydra/examples/experimental/outputs/2022-03-16/14-51-30
+[2022-03-16 14:59:21,666][__main__][INFO] - cfg.foo=bar
+```
+You will notice `my_app.log` is updated with the logging from the second run, but Callbacks are not called this time. Read on to learn more.
+
+
+### Important Notes
+This is an experimental feature. Please reach out if you have any question. 
+- Only single run is supported.
+- `--experimental-rerun` cannot be used with other command-line options or overrides. They will simply be ignored.
```

Comment:
We should at least warn about ignored args.

Reply:
edited the warning message in [b6ae6ed](https://github.com/facebookresearch/hydra/pull/2098/commits/b6ae6eda6d2a7d8a68a6ffda221acfb40cc96a95)